helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -531,6 +531,8 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
       // statusDeploymentGroupRemovedHosts may not exist for deployment groups created before it was
       // introduced.
       client.ensurePath(Paths.statusDeploymentGroupRemovedHosts(groupName));
+      client.ensurePathAndSetData(Paths.statusDeploymentGroupRemovedHosts(groupName),
```

Comment:
I think if the path is created on the line above, then this call will be a no-op and you won't write the `[]`